connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/Observation.scala

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql
+
+import java.util.UUID
+
+class Observation(name: String) extends ObservationBase(name) {
```

Comment:
Why do we need this subclass?

Reply:
Oh true, this one is not needed. We could use ObservationBase directly.

Reply:
The purpose of this `Observation` class is to provide a familiar name for users, avoiding the need to learn a new one. This is necessary because all of spark-core, spark-api, and spark-connect-client share the same class path, thus we must distinct the parent class `ObservationBase` from others.

Reply:
Yes, we have to call it Observation. Naming it anything else will break compatibility. However my question is more about why we need a subclass? Is there 'frontend' specific code involved?